tests/userController.test.js

```diff
@@ -100,4 +100,71 @@ describe("user controller", () => {
       });
     });
   });
-});
+
+  describe("updates a user", () => {
+      it("updates an existent user", async() => {
+        const req = {
+          body: { id: 1, email: "newemail@example.com", password: "newpassword", permission: "newpermission" }
+        };
+        const res = {
+          status: jest.fn().mockReturnThis(),
+          send: jest.fn()
+        };
+    
+        database.User.update = jest.fn().mockResolvedValue([1]);
+    
+        await updateUser(req, res);
+    
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledWith("User data updated");
+      });
+      it("handles a request to update non existent user", async() => {
+          const req = {
+            body: { id: 99, email: "newemail@example.com", password: "newpassword", permission: "newpermission" }
+          };
+          const res = {
+            status: jest.fn().mockReturnThis(),
+            send: jest.fn()
+          };
+          
+          // O primeiro elemento da função updated do sequeleize é o número de linhas afetadas por essa atualização
```

Comment:
1. English please.
2. O que e `O primeiro elemento da função updated do sequeleize`? 
3. Nao entendi nenhum dos comentarios. 
4. sequeleize esta escrito errado.

Maybe update the test to:

`// Mock update to return 0 updated rows`

Reply:
Fiz um comentário anotando para mim mesmo que o primeiro elemento [0] consta as mudanças realizas pela função update do sequelize, você gostaria que eu o removesse?